src/features/filter.ts

```diff
@@ -8,5 +8,18 @@ const initialState = {
 export const filterSlice = createSlice({
   name: 'filter',
   initialState,
-  reducers: {},
+  reducers: {
+    setQuery: (state, action: PayloadAction<string>) => ({
+      ...state,
+      query: action.payload,
+    }),
+    setStatus: (state, action: PayloadAction<string>) => ({
```

Comment:
The `setStatus` action uses `PayloadAction<string>`. Ensure that this aligns with the `Status` type used in other parts of the application. If `Status` is a specific type or enum, consider using `PayloadAction<Status>` to maintain type consistency.